docs/user/using-an-OIDC-provider.md

```diff
@@ -0,0 +1,162 @@
+# Using an OIDC Provider with Kubeapps
+
+OpenID Connect (OIDC) is a simple identity layer on top of the OAuth 2.0 protocol, which allows computing clients to verify the identity of an end-user based on the authentication performed by an authorization server, as well as to obtain basic profile information about the end-user.
+
+It is possible to configure your Kubernetes cluster to use an OIDC provider in order to manage accounts, groups and roles with a single application. This guide will explain how you can use an existing OIDC provider to authenticate users within Kubeapps.
+
+## Pre-requisites
+
+For this guide we assume that you have a Kubernetes cluster that is properly configured to use an Identity Provider to handle the authorization of your cluster. You can find more information about how Kubernetes uses OIDC tokens [here](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens). This means that the Kubernetes API server should be configured to use that OIDC provider.
+
+There are several Identity Providers (IdP) that can be used in a Kubernetes cluster. The steps of this guide have been validated using the following providers:
+
+- [Keycloak](https://www.keycloak.org/): Open Source Identity and Access Management.
+- [Dex](https://github.com/dexidp/dex): Open Source OIDC and OAuth 2.0 Provider with Pluggable Connectors.
+- [Azure Active Directory](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis): Identity Provider that can be used for AKS.
+- [Google OpenID Connect](https://developers.google.com/identity/protocols/OpenIDConnect): OAuth 2.0 for Google accounts.
+
+For Kubeapps to use an Identity Provider it's necessary to configure at least the following parameters:
+
+- Client ID: Client ID of the IdP.
+- Client Secret: (If configured) Secret used to validate the Client ID.
+- OIDC Issuer URL (or discovery URL): URL of the OIDC issuer.
+
+**Note**: Depending on the configuration of the Identity Provider more parameters may be needed.
+
+In the next sections we will explain how you can find the parameters above for some of the identity providers tested.
```

Comment:
Is this necessary? We're assuming they've already configured their Kubernetes clusters correctly, and so should already know these values.